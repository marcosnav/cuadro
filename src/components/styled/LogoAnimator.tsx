import styled, { css, keyframes, StyledProps } from 'styled-components';

interface IProps {
  animate?: boolean;
}

const rotate = keyframes`
  0%, 45% {
    transform: rotate(0deg);
  }
  90%, 100% {
    transform: rotate(540deg);
  }
`;

const wrapAnimation = (props: StyledProps<IProps>) => (
  props.animate ? css`${rotate} 2s ease infinite;` : 'none;'
);

const LogoAnimator = styled.div<IProps>`
  animation: ${(props) => wrapAnimation(props)}
  height: 20px;
  position: relative;
  width: 20px;
`;

LogoAnimator.displayName = 'LogoAnimator';

export default LogoAnimator;
