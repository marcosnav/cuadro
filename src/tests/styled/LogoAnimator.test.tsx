import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import LogoAnimator from './../../components/styled/LogoAnimator';

describe('<LogoAnimator />', () => {
  test('renders properly', () => {
    const component = mount(<LogoAnimator />);
    expect(component).toHaveStyleRule('animation', 'none');
    expect(component).toHaveStyleRule('margin-top', '0');
    expect(component).toMatchSnapshot();
  });

  test('animate: true', () => {
    const component = mount(<LogoAnimator animate={true} />);
    expect(component).toHaveStyleRule('animation', expect.stringContaining('2s ease infinite'));
    expect(component).toHaveStyleRule('margin-top', 'calc(50vh - 98px)');
    expect(component).toMatchSnapshot();
  });
});
