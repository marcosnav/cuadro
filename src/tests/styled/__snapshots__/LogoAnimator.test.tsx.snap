// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoAnimator /> animate: true 1`] = `
.c0 {
  -webkit-animation: lcQkPt 2s ease infinite;
  animation: lcQkPt 2s ease infinite;
  height: 20px;
  margin-top: calc(50vh - 98px);
  position: relative;
  -webkit-transition: margin-top 0.6s ease;
  transition: margin-top 0.6s ease;
  width: 20px;
}

<LogoAnimator
  animate={true}
>
  <div
    className="c0"
  />
</LogoAnimator>
`;

exports[`<LogoAnimator /> renders properly 1`] = `
.c0 {
  -webkit-animation: none;
  animation: none;
  height: 20px;
  margin-top: 0;
  position: relative;
  -webkit-transition: margin-top 0.6s ease;
  transition: margin-top 0.6s ease;
  width: 20px;
}

<LogoAnimator>
  <div
    className="c0"
  />
</LogoAnimator>
`;
